src/BlueprintParser.h

```diff
@@ -430,9 +508,9 @@ namespace snowcrash {
                 mson::NamedTypeInheritanceTable::iterator inhIt = pd.namedTypeInheritanceTable.find(superType);
 
                 // Check for recursive MSON definitions
-                if (inhIt == pd.namedTypeInheritanceTable.end() || inheritanceNodes.find(inhIt->second.first) != inheritanceNodes.end()) {
+                if (inhIt == pd.namedTypeInheritanceTable.end()) {
```

Comment:
So, since this condition is loosened – can't this lead into additional problems?